src/combinatorics/simple_graph/connectivity.lean

```diff
@@ -261,6 +292,25 @@ by { subst_vars, refl }
 | _ _ _ nil _ := by simp
 | _ _ _ (cons _ _) _ := by simp [length_append, add_left_comm, add_comm]
 
+@[simp] lemma length_concat {u v w : V} (p : G.walk u v) (h : G.adj v w) :
+  (p.concat h).length = p.length + 1 := length_append _ _
+
+lemma cons_eq_concat.length {u v w : V} (h : G.adj u v) (p : G.walk v w) :
+  (cons_eq_concat h p).2.1.length = p.length :=
+begin
+  induction p generalizing u,
+  { refl },
+  { generalize : cons_eq_concat h (cons p_h p_p) = res,
+    obtain ⟨x, a, b, res⟩ := res,
+    specialize p_ih p_h,
+    generalize_hyp : cons_eq_concat p_h p_p = res' at p_ih,
+    obtain ⟨y, c, d, res'⟩ := res',
+    have res2 := congr_arg length res,
+    have res'2 := congr_arg length res',
+    simp only [length_cons, length_concat, add_left_inj] at res2 res'2 ⊢,
+    rw [res'2, p_ih, ← res2], }
+end
```

Comment:
This proof does not need to be so long ...
```suggestion
by simpa only [length_cons, length_concat, add_left_inj]
  using congr_arg length (cons_eq_concat h p).2.2.2.symm
```

Reply:
It's not like I went ought of my way to make a long proof :wink:

Thanks for finding a shorter one.